website/static/js/filepage/index.js

```diff
@@ -215,7 +216,58 @@ var FileViewPage = {
                 m('button.btn.btn-sm.btn-danger.file-delete', {onclick: $(document).trigger.bind($(document), 'fileviewpage:delete')}, 'Delete')
             ]) : '',
             m('.btn-group.m-t-xs', [
-                m('button.btn.btn-sm.btn-primary.file-download', {onclick: $(document).trigger.bind($(document), 'fileviewpage:download')}, 'Download')
+                m('#sharebutton.btn.btn-sm.btn-primary.file-share', {onclick: function () {
+                    var link = $('iframe').attr('src');
+                    var height = $('iframe').attr('height');
+                    link = link.substring(0, link.indexOf('download') + 8) + '%26mode=render';
+                    var url = link.substring(0, link.indexOf('render'));
+                    m.render(document.getElementById('popOver'), [
+                        m('ul.nav.nav-tabs.nav-justified', [
+                            m('li.active', m('a[href="#share"][data-toggle="tab"]', 'Share')),
+                            m('li', m('a[href="#embed"][data-toggle="tab"]', 'Embed'))
+                        ]), m('br'),
+                        m('.tab-content', [
+                            m('.tab-pane.fade.in.active#share', m('.input-group', [
+                                m('span.input-group-btn', m('button#copyBtn.btn.btn-default.btn-md[type="button"][style="height: 34px"][data-clipboard-text="' + link + '"]', m('.fa.fa-copy'))),
+                                m('input.form-control[readonly][type="text"][value="'+ link +'"]')
+                            ])),
+                            m('.tab-pane.fade#embed', [
+                                m('p', 'Dynamically Render iFrame with JavaScript'),
+                                m('textarea.form-control[readonly][type="text"][value="' +
+                                    '<script>window.jQuery || document.write(\'<script src="//code.jquery.com/jquery-1.11.2.min.js">\\x3C/script>\') </script>'+
```

Comment:
After talking to @chrisseto this seemed like the best option to allow the loading mfr symbol to work correctly.